libnd4j/include/ops/declarable/helpers/impl/ctcBeam.cpp

```diff
@@ -646,7 +647,7 @@ beamSearch_(const NDArray& logit, const NDArray& sequence_length, NDArray& resul
     auto batch_stride = rank > 2 ? strides[0] : 0;
     auto inc_p = strides[rank - 2];
     auto element_stride = logit.stridesOf()[rank - 1];
```

Comment:
Mind adding comments to each of these so someone knows why this was called? Mainly for external contributors or people just reading the implementation.

Reply:
my bad, I do not know why I did not put it as a derived datatype for the future. that is a lesser confusion.
 
we have a CTC loss (maybe in the future, cudnn-they consider other types as well) restricted (only for Float32 and input32). I'm thinking of refactoring ctc Loss again ( it needs if def sections and proper comments so it was not confusing ). and also adding v8 api calls .

Reply:
I refactored those (syncToHost calls) by adding them inside NDArray::preparePrimaryUse.  and commented